### ISSUE TITLE
Compatibility with Linked Data Platform 1.0

### ISSUE THREAD

**Issue by muratseyhan** [jasnell/w3c-socialwg-activitystreams#111]
Many thanks to all WG members for Activity Streams 2.0. It's been only a few days since I discovered this, and I have a few quite related questions that I wasn't able to find an answer in the specification nor the previous issues. I apologize if this is stupid or was discussed before.

Activity Streams 2.0 does not seem to be compatible with Linked Data Platform 1.0. There might be others, but a basic conflict is the following.
  * [Activity Streams 2.0](http://jasnell.github.io/w3c-socialwg-activitystreams/activitystreams2.html) dictates the MIME type `application/activity+json` for documents conforming to the Activity Streams 2.0 format.
  * [LDP 1.0](http://www.w3.org/TR/ldp/) dictates `application/ld+json` or `text/turtle` depending on the `Accept` header in the request. Conformance of the server with LDP 1.0 for specific resources is cued with a link header entry, such as `Link: <http://www.w3.org/ns/ldp#Resource>; rel="type"`. 

So my questions are:

*  I have noticed a discussion on MIME types `application/ld+json` and `application/activity+json`, but what about other serialization formats?

The specification provides examples in some other serialization formats as well. Are those examples not AS 2.0 documents?  

* Why is Activity Streams 2.0 not intended to be compatible with Linked Data Platform 1.0?

LDP servers are actually allowed to serve non-LDP resources as long as they advertise their LDP resources properly, but the current state of the specifications prohibits an AS 2.0 document to be served as an LDP resource. Also, an LDP server publishing RDF representations as non-LDP resources sounds quite weird. 

Maybe the use cases for LDP and AS are mutually exclusive, but I cannot think of a reason why it should be. AS provides a representation format for the resources that the activities are performed on (along with the activities), and one could wish to publish those resources as LDPRs.

I think AS could even extend LDP, and benefit from its solutions to their common problems, such as representation of collections or pagination. In that case, each AS object would be an LDP resource, and could be advertised with a Link header entry, such as:

    Link: <http://www.w3.org/ns/activitystreams#Object>; rel="type", 
             <http://www.w3.org/ns/ldp#Resource>; rel="type" 

Apologies for long message and the heap of questions. Thanks in advance!

**Comment by akuckartz** [comment]
muratseyhan wrote:
>   * I have noticed a discussion on MIME types |application/ld+json| and
>     |application/activity+json|, but what about other serialization formats?
> 
> The specification provides examples in some other serialization formats
> as well. Are those examples not AS 2.0 documents?

The WG charter requires a JSON serialization. Other serializations
certainly can be used, but their specification is not in scope for the
WG. But the WG uses JSON-LD and I think that one can therefore say that
other RDF-serializations are therefore implicitly specified.

>   * Why is Activity Streams 2.0 not intended to be compatible with
>     Linked Data Platform 1.0?
> LDP servers are actually allowed to serve non-LDP resources as long as
> they advertise their LDP resources properly, but the current state of
> the specifications prohibits an AS 2.0 document to be served as an LDP
> resource.

@muratseyhan Can you please elaborate on that? How is that prohibited?

**Comment by muratseyhan** [comment]
Thanks for your reply, @akuckartz! 

Now that I checked LDP specification again, that statement seems to be wrong. I used the term `LDP resource` for [LDP-RS](http://www.w3.org/TR/ldp/#dfn-linked-data-platform-rdf-source), and `non-LDP resource` for [LDP-NR](http://www.w3.org/TR/ldp/#dfn-linked-data-platform-non-rdf-source). LDP-RSs are served with Content-Type headers `application/ld+json` and `text/turtle` for JSON-LD and turtle, however they are apparently allowed to have additional representations along with those. This seems to allow serving JSON-LD also with `application/activity+json`.

I am closing the issue as I received a satisfactory explanation.
